lib/tree.js

```diff
@@ -12,40 +12,115 @@ class Tree {
         this.root = null;
     }
 
-    // Time Complexity: ?
-    // Space Complexity: ?
+    // Time Complexity: O(log n)
+    // Space Complexity: O(log n)
     add(key, value) {
-        throw new Error("This method hasn't been implemented yet!");
+        const newNode = new TreeNode(key, value);
+        if (!this.root) {
+            this.root = newNode;
+        } else {
+            this.addHelper(this.root, newNode);
+        }
     }
+    
+    addHelper(current, newNode) {
+        if (!current) {
+            return newNode;
+        } else {
+            if (current.key > newNode.key) {
+                current.left = this.addHelper(current.left, newNode);
+            } else if (current.key <= newNode.key) {
+                current.right = this.addHelper(current.right, newNode);
+            }
+        }
 
-    // Time Complexity: ?
-    // Space Complexity: ?
+        return current;
+    }
+
+    // Time Complexity: O(log n)
+    // Space Complexity: O(log n)
     find(key) {
```

Comment:
👍 The space/time complexity is O(log n) if the tree is balanced and O(n) if it's not.